extensions-core/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java

```diff
@@ -173,9 +172,17 @@ public InputStream getBlockBlobInputStream(long offset, Long length, final Strin
   public void batchDeleteFiles(String containerName, Iterable<String> paths, Integer maxAttempts)
       throws BlobBatchStorageException
   {
-
-    BlobBatchClient blobBatchClient = new BlobBatchClientBuilder(getOrCreateBlobContainerClient(containerName, maxAttempts)).buildClient();
-    blobBatchClient.deleteBlobs(Lists.newArrayList(paths), DeleteSnapshotsOptionType.ONLY);
+    BlobContainerClient blobContainerClient = getOrCreateBlobContainerClient(containerName, maxAttempts);
+    List<String> blobUris = Streams.stream(paths).map(path -> blobContainerClient.getBlobContainerUrl() + "/" + path).collect(Collectors.toList());
+
+    // We have to call forEach on the response because this is the only way azure batch will throw an exception on a operation failure.
+    azureClientFactory.getBlobBatchClient(blobContainerClient).deleteBlobs(
+        blobUris,
+        DeleteSnapshotsOptionType.INCLUDE
+    ).forEach(response ->
```

Comment:
Should we error / warning log any errors that are found here, or is that done at a lower level?